### PR TITLE
[INLONG-6176][Agent] Support collect data from Oracle

### LINKED ISSUE #6176: [Feature][Agent] Support collect data from Oracle
### Description

you can refer to:
https://inlong.apache.org/docs/next/design_and_concept/how_to_write_plugin_agent

support version: 11,12,19 

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haibo-duan**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #6176 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by dockerzhang** [comment]
@haibo-duan here we need to add Oracle Source to the JobProfileDto.class. You can refer to MongoDB at https://github.com/apache/inlong/pull/5674.

maybe other sources you added in the agent also add the job definition, like PostgreSQL/SQLServer.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/OracleReader.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    @Override
+    public boolean isSourceExist() {
+        return true;
+    }
+
+    @Override
+    public void init(JobProfile jobConf) {
+        super.init(jobConf);
+        int batchSize = jobConf.getInt(JOB_DATABASE_BATCH_SIZE, DEFAULT_JOB_DATABASE_BATCH_SIZE);
+        String userName = jobConf.get(JOB_DATABASE_USER);
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/OracleReader.java:171]: finalize string

**Review by EMsnap — COMMENTED** [review]
As for now, the implementation is more like a sql source.
Oracle source should be implemented as a online source just like mysql source

**Review by EMsnap — COMMENTED** [review]
If u want a sql source for oracle please try to enhance the sql source ability for oracle directly

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/OracleReader.java [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+                    }
+                }).build();
+
+        executor = Executors.newSingleThreadExecutor();
+        executor.execute(engine);
+
+        LOGGER.info("get initial snapshot of job {}, snapshot {}", instanceId, getSnapshot());
+    }
+
+    private String serializeOffset() {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/OracleReader.java:259]: DUPLICATE CODE IN MONGODB AND PG, CAN YOU abstract this?

**Review by pocozh — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/OracleReader.java [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import static org.apache.inlong.agent.constant.CommonConstants.DEFAULT_MAP_CAPACITY;
+import static org.apache.inlong.agent.constant.CommonConstants.PROXY_KEY_DATA;
+
+/**
+ * Read data from Oracle database by SQL
+ */
```
Comment (GanfengTan) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/OracleReader.java:63]: Please, update this doc

**Review by healchow — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**